Sources/UIApplication.swift

```diff
@@ -97,3 +97,33 @@ public func nativeProcessEventsAndRender(env: UnsafeMutablePointer<JNIEnv>, view
     CFRunLoopRunInMode(kCFRunLoopDefaultMode, max(0.001, remainingFrameTime / 2), true)
 }
 #endif
+
+extension UIApplication {
+    public static let didEnterBackgroundNotification
```

Comment:
Ok interesting, didn't realise these were available publicly on iOS. I noticed on iOS they're implemented as `class let` rather than `static let` -> I think the subtle difference could be that `class let`s can be overridden in a subclass (but I'm not certain about that). Which might let you override the notifications that get sent by the system (for whatever crazy reason).

If that's the case, we should change `.post(UIApplication.didBecomeActiveNotification)` to `.post(didBecomeActiveNotification)` to take possible subclassing into account. We may need to make `onDidEnterBackground` etc. `class` (instead of `static`) functions too for this to work.

To be honest though I'm a bit fuzzy on the minor differences – what I'm saying here may all be meaningless. But I do remember running into this issue with `UIView.layerClass` and from memory changing it to `class var` made the intended behaviour (the ability to change layerClass in subclasses) possible.

Reply:
Yes, you're right - `class` basically means that sub-classes can "override" the variable or function, while `static` makes that property/function implicitly `final`.
Updating these :)